soc/silabs/silabs_s2/soc.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief SoC initialization for Silicon Labs Series 2 products
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <em_chip.h>
+
+#include <sl_device_init_dcdc.h>
+#include <sl_clock_manager_init.h>
+
+#ifdef CONFIG_PM
+#include <sl_hfxo_manager.h>
+#include <sl_power_manager.h>
+#endif
```

Comment:
Probably it does not hurt to include `sl_hfxo_manager.h` and `sl_power_manager.h` even if `!CONFIG_PM`.